src/cli/main.ts

```diff
@@ -39,7 +39,7 @@ async function guardNotGlobalCLIWithLocalProject(
   // TODO data is attainable from layout scan calculated later on... not optimal to call this twice...
   const projectType = await Layout.scanProjectType({ cwd: process.cwd() })
 
-  if (projectType.type === 'NEXUS_project' && isProcessFromProjectBin(projectType.packageJsonLocation.path)) {
```

Comment:
Based on the feedback https://github.com/graphql-nexus/nexus/issues/526#issuecomment-613444223 and just thinking about it now...

We should probably tightly couple the error message and the checks we're making internally. For example the internal checks are using `realpathSync` while we're rebuilding paths for the error messages manually. I fear this is asking for trouble.

Some ideas on how to solve:

- assertion function that encapsulates all the logic
- validation function that returns `CustomError | void`. `void` means no problem. `CustomError` includes structured data `{ projectBinPath, executionBinPath }`; Then this is interpolated into a rich fatal message.

I think my preference is the latter. Thoughts?